hudi-common/src/main/java/org/apache/hudi/common/table/log/block/HoodieHFileDataBlock.java

```diff
@@ -135,8 +138,13 @@ protected byte[] serializeRecords(List<HoodieRecord> records) throws IOException
       }
 
       final byte[] recordBytes = serializeRecord(record, writerSchema);
-      ValidationUtils.checkState(!sortedRecordsMap.containsKey(recordKey),
-          "Writing multiple records with same key not supported for " + this.getClass().getName());
+      if (sortedRecordsMap.containsKey(recordKey)) {
+        LOG.error("Found duplicate record with recordKey: " + recordKey);
```

Comment:
hfile allows duplicated to be added as long as they are sorted.

Reply:
that's right but it's better to drop duplicates.

Reply:
I see that `HoodieAvroHFileWriter` does not throws when it encounters duplicates if `this.hfileConfig.allowDuplicatesToBeInserted()` returns true, so that option is only effective for the writer scope then?